README.md

```diff
@@ -396,40 +516,210 @@ Make sure that you have specified `authenticationEndpoint` during SDK initializa
+| customMetadata          | Map<String, Any>                                                                          | Optional. Key-value data to be associated with the uploaded file. Check `overwriteCustomMetadata` parameter to understand default behaviour. Before setting any custom metadata on an asset you have to create the field using [custom metadata fields API](https://docs.imagekit.io/api-reference/custom-metadata-fields-api).                                                    |
+| policy                  | [UploadPolicy](README.md#UploadPolicy)                                                    | Optional. Set the custom policy to override the default policy for this upload request only. This doesn't modify the default upload policy.                                                                                                                                                                                                                                        |
+| preprocess              | [ImagePreprocess](README.md#ImagePreprocess)/[VideoPreprocess](README.md#VideoPreprocess) | Optional. Set the set the parameters for preprocessing the image/video before uploads. This doesn't modify the default upload policy.                                                                                                                                                                                                                                              |
+| imageKitCallback        | [ImageKitCallback](imagekit/src/main/java/com/imagekit/android/ImageKitCallback.kt)       | Required.                                                                                                                                                                                                                                                                                                                                                                          |
 
 Sample Usage
 ```kotlin
 ImageKit.getInstance().uploader().upload(
-  file = image,
-  fileName = "sample-image.jpg",
-  useUniqueFilename = true,
-  tags = ["demo"],
-  folder = "/",
-  isPrivateFile =  false,
-  customCoordinates = "",
-  responseFields = "",
-  imageKitCallback = object: ImageKitCallback {
+    file = image,
+    fileName = "sample-image.jpg",
+    useUniqueFilename = true,
+    tags = ["demo"],
+    folder = "/",
+    isPrivateFile =  false,
+    customCoordinates = "",
+    responseFields = "",
+    extensions = listOf(mapOf("name" to "google-auto-tagging", "minConfidence" to 80, "maxTags" to 10)),
```

Comment:
Is there not a better way to take this information? By better I mean more strongly typed like we do all things in Java/Kotlin.

Reply:
This is pending. Please share alternative syntax.

Reply:
The alternative syntax for `extensions` argument, with `JsonArray` as parameter type would be like this:
```
//Kotlin
...,
extensions = JSONArray(arrayOf(
    JSONObject().apply {
        put("name", "google-auto-tagging")
        put("minConfidence", 80)
        put("maxTags", 10)
    }
))
```